rules/windows/process_creation/Possible Manipulation Of Tokens on a Windows computers remotely Detected via Impersonate.yml

```diff
@@ -0,0 +1,29 @@
+title: Possible Manipulation Of Tokens on a Windows computers remotely Detected via Impersonate
+id: cf0c254b-22f1-4b2b-8221-e137b3c0af94
+status: experimental
+description: Threat actors leverage a standalone binary (Impersonate/) that you can use to manipulate tokens on a Windows computers remotely (PsExec/WmiExec) or interactively
+references:
+    - https://sensepost.com/blog/2022/abusing-windows-tokens-to-compromise-active-directory-without-touching-lsass/
+author: Sai Prashanth Pulisetti @pulisettis
+date: 2022/12/21
+modified: 2022/12/21
+tags:
+    - attack.command_and_control
+    - attack.lateral_movement
+logsource:
+    product: windows
+    category: process_creation
```

Comment:
process_creation don't have HostApplication field.
And the is no `HostApplication` in the ref , do you have a example of the log ?
Thanks

Reply:
Hey @frack113  I took the reference from the logs that I tested in my lab. 

Please see below example log : 

Process Create:
RuleName: -
UtcTime: 2022-12-20 10:06:46.212
ProcessGuid: {636d729a-2fe2-5c98-1adf-000000004b00}
ProcessId: 19281
Image: C:\Windows\Temp\Impersonate.exe
FileVersion: -
Description: -
Product: -
Company: -
OriginalFileName: -
CommandLine: C:\Windows\Temp\Impersonate.exe  exec 5 "net group allow_usb /domain"   
CurrentDirectory: C:\WINDOWS\system32\
User: NT AUTHORITY\SYSTEM
LogonGuid: {636d729a-2fe2-5c98-1adf-000000004b00}
LogonId: 0x3E7
TerminalSessionId: 0
IntegrityLevel: System
Hashes: MD5=9520714AB576B0ED01D1513691377D01,SHA256=E81CC96E2118DC4FBFE5BAD1604E0AC7681960143E2101E1A024D52264BB0A8A,IMPHASH=0A358FFC1697B7A07D0E817AC740DF62
ParentProcessGuid: {729a5c98-2fe2-636d-18df-000000004b00}
ParentProcessId: 8172
ParentImage: C:\Windows\System32\cmd.exe
ParentCommandLine: C:\WINDOWS\system32\cmd.exe  /Q /c C:\WINDOWS\TEMP\YuiSk.bat 
ParentUser: NT AUTHORITY\SYSTEM

Reply:
`CommandLine: C:\Windows\Temp\Impersonate.exe exec 5 "net group allow_usb /domain"` is what you are looking for
```yaml
    selection_exe:
        CommandLine|contains: 'impersonate.exe'
    selection_opt:
        CommandLine|contains:
            - ' list '
            - ' exec '
            - ' token '
            - ' imp_exe '
    condition: all of selection_*
```
I have add space to the verb  in selection_opt

Reply:
We can add Hashes too
```yaml
    selection_commandline_exe:
        CommandLine|contains: 'impersonate.exe'
    selection_commandline_opt:
        CommandLine|contains:
            - ' list '
            - ' exec '
            - ' token '
            - ' imp_exe '
    selection_hash_plain:
        Hashes|contains:
            - 'MD5=9520714AB576B0ED01D1513691377D01'
            - 'SHA256=E81CC96E2118DC4FBFE5BAD1604E0AC7681960143E2101E1A024D52264BB0A8A'
            - 'IMPHASH=0A358FFC1697B7A07D0E817AC740DF62'
    selection_hash_ext:
        - MD5: '9520714AB576B0ED01D1513691377D01'
        - SHA256: 'E81CC96E2118DC4FBFE5BAD1604E0AC7681960143E2101E1A024D52264BB0A8A'
        - IMPHASH: '0A358FFC1697B7A07D0E817AC740DF62'
    condition: all of selection_commandline_* or 1 of selection_hash_*
```

Reply:
Thanks, @frack113 . Yes, that makes sense to me. Since I didn't add hashes because they may change if they modify the code and compile again. But helpful to have.

Reply:
In my opinion, the name of the binary will change faster than the hash. So it's important in this case to have the hashes since the detection is based solely on the binary name (as the CLI flags cannot be used on their own to identify the tool).

So in this case it's not only helpful it's necessary :)

Reply:
Completely agreed!

Reply:
Changes has been made and let me know if anything is pending from my end :-)